src/main/java/org/candlepin/dto/api/v1/ConsumerTranslator.java

```diff
@@ -137,8 +138,16 @@ public ConsumerDTO populate(ModelTranslator translator, Consumer source, Consume
                 dest.setOwner(owner != null ? translator.translate(owner, NestedOwnerDTO.class) : null);
             }
 
-            Environment environment = this.environmentCurator.getConsumerEnvironment(source);
-            dest.setEnvironment(translator.translate(environment, EnvironmentDTO.class));
+            if (source.getEnvironmentIds() != null) {
+                Collection<String> envIDs = source.getEnvironmentIds().values();
+                for (String id : envIDs) {
+                    dest.addEnvironments(translator
+                        .translate(this.environmentCurator.listById(id), EnvironmentDTO.class));
+                }
+            }
+            else {
+                dest.setEnvironments(Collections.emptyList());
```

Comment:
Is the specification such that we always output a list, or is null a valid value for the environments property?

Reply:
I don't think this was thought of during the specification, and looking at our various translators we're very inconsistent with what we do about collections (sometimes we set to null, sometimes empty collection). It looks like most of the time we set to null, so we should try to standardize that way?

Reply:
Done.